### PR TITLE
Add lnbits app

### PR DESCRIPTION
**Pull request by louneskmt**
# App Submission

### App name
LNbits

### Version
v0.2.0

### One line description of the app
_(max 50 characters)_

Multi-user and wallet system for Lightning Network

### Summary of the app
_(50 to 200 words)_

LNbits is a simple multi-user and account system for Lightning Network that can be used for creating separate Lightning wallets for friends and family members. You can also create multiple accounts for yourself to mitigate the risk of exposing applications to your full balance via unique API keys for each wallet.

LNbits is packaged with tools to help manage funds, such as a table of transactions, line chart of spending, export to CSV, and more to come. It provides an extendable platform for expanding Lightning Network functionality via LNbits extension framework, and can also be used as a fallback wallet for the LNURL scheme.

### Developed by
LNbits

### Developer website

https://lnbits.org

### Source code repository

https://github.com/lnbits/lnbits

### Support link
_(Link to your Telegram support channel, GitHub issues/discussions, support portal, or any other place where users could contact you for support.)_

https://github.com/lnbits/lnbits/issues/new

### Uses
- [ ] Bitcoin Core
- [ ] Electrum server
- [x] LND

### 256x256 SVG icon
_(GitHub doesn't allow uploadig SVGs directly. Upload your file to an alternate service, like https://svgur.com, and paste the link below.)_

icon

### App screenshots
_(Upload 3 to 5 high-quality screenshots (at least 1280x800px) of your app in PNG format.)_

image
image
image

### I have tested my app on:
- [ ] [Umbrel dev environment](https://github.com/getumbrel/umbrel-dev)
- [x] [Umbrel OS on a Raspberry Pi 4](https://github.com/getumbrel/umbrel-os)
- [ ] [Custom Umbrel install on Linux](https://github.com/getumbrel/umbrel#-installation)

### CODE REVIEW

**Review by mayankchhabra — DISMISSED** [review]
Really nice job on packaging LNbits, @louneskmt! Works flawlessly.

**Comment by louneskmt** [comment]
Source: https://github.com/louneskmt/docker-lnbits

**Comment by mayankchhabra** [comment]
cc @arcbtc: LNbits looks ready to be shipped in the next Umbrel release! Can you please quickly eyeball this once?

Also, everything in the PR description above is what the user will see in the LNbits app description on the Umbrel App Store, so feel free to suggest any edits if you'd like.

File: apps/lnbits/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      LNBITS_BACKEND_WALLET_CLASS: LndWallet
+      LND_GRPC_ENDPOINT: $LND_IP
+      LND_GRPC_PORT: $LND_GRPC_PORT
+      LND_GRPC_CERT: "/lnd/tls.cert"
+      LND_GRPC_MACAROON: "/lnd/data/chain/bitcoin/$BITCOIN_NETWORK/admin.macaroon"
+
+      # App
+      LNBITS_SITE_TITLE: "LNbits - Umbrel"
+      LNBITS_DEFAULT_WALLET_NAME: "LNbits wallet"
+      LNBITS_DISABLED_EXTENSIONS: "amilk"
```
Comment (lukechilds) [inline apps/lnbits/docker-compose.yml:36]: Why is this extension disabled?

Comment (louneskmt) [reply apps/lnbits/docker-compose.yml:36]: It allows an user to automatically claim sats from an LNURLw. For example, a LNURL to claim 1000 sats every minute. I found this pretty unfair as the delay is here to allow more people to claim sats (e.g. a test LNURL in a Bitcoin bar, for newbies to discover Lightning) and not just have a single person claiming the whole balance. 

If necessary I can enable it.

Comment (lukechilds) [reply apps/lnbits/docker-compose.yml:36]: Hmmn, I agree that doesn't sound like very nice behaviour but I'm not sure it should be our decision to blacklist it.

It's not doing anything bad by default, right? A user would have to enable manually enable it?

Comment (louneskmt) [reply apps/lnbits/docker-compose.yml:36]: > A user would have to manually enable it?

Each extension has to be activated from the extensions page.

File: apps/lnbits/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      LNBITS_ALLOWED_USERS: ""
```
Comment (lukechilds) [inline apps/lnbits/docker-compose.yml:37]: Why is this empty?

Are there any docs for these settings? I had a quick look at the LNbits repo but didn't find any.

Comment (louneskmt) [reply apps/lnbits/docker-compose.yml:37]: I used the default values here: https://github.com/lnbits/lnbits/blob/master/.env.example

**Comment by lukechilds** [comment]
IS the app being built in development mode?

https://github.com/louneskmt/docker-lnbits/blob/4767453412e8c70249940586fe5cddc0f0fbb396/Dockerfile#L10

**Comment by louneskmt** [comment]
> Is the app being built in development mode?
> 
> 
> 
> https://github.com/louneskmt/docker-lnbits/blob/4767453412e8c70249940586fe5cddc0f0fbb396/Dockerfile#L10

I don't think so, this is the default value. Maybe LNbits needs Quark in dev mode, idk. 

https://lnbits.org/guide/installation.html

File: apps/lnbits/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+x-logging:
+  &default-logging
+  driver: journald
+  options:
+    tag: "umbrel-app {{.Name}}"
+
+services:
+  web:
+    image: louneskmt/lnbits:v0.2.0
```
Comment (lukechilds) [inline apps/lnbits/docker-compose.yml:11]:
```suggestion
    image: louneskmt/lnbits:v0.2.0@sha256:15e10d926d4b14774c7b2cc74f722e887ec198f1f1e7d35ee66a517e297ae064
```

File: apps/lnbits/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      LND_GRPC_PORT: $LND_GRPC_PORT
+      LND_GRPC_CERT: "/lnd/tls.cert"
+      LND_GRPC_MACAROON: "/lnd/data/chain/bitcoin/$BITCOIN_NETWORK/admin.macaroon"
+
+      # App
+      LNBITS_SITE_TITLE: "LNbits - Umbrel"
+      LNBITS_DEFAULT_WALLET_NAME: "LNbits wallet"
+      LNBITS_DISABLED_EXTENSIONS: "amilk"
+      LNBITS_ALLOWED_USERS: ""
+      LNBITS_FORCE_HTTPS: "true"
```
Comment (lukechilds) [inline apps/lnbits/docker-compose.yml:38]: Do you know what this option does?

It doesn't appear to be using HTTPS at all for me.

File: apps/lnbits/docker-compose.yml (hunk 1 above)
Comment (louneskmt) [reply apps/lnbits/docker-compose.yml:36]: But of the extension is disabled in this env var, it cannot be installed from the web page. So yeah, if it shouldn't be our decision to blacklist it, we should remove this line.

File: apps/lnbits/docker-compose.yml (hunk 4 above)
Comment (louneskmt) [reply apps/lnbits/docker-compose.yml:38]: Not really. I think we can get ride of this 3 lines (disabled extensions, force http and allowed users).

**Review by lukechilds — APPROVED** [review]
Looking good, nice job!

**Comment by lukechilds** [comment]
No point blocking the merge for this but the Docker image is very large:

```
$ docker images louneskmt/lnbits:v0.2.0
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
louneskmt/lnbits    v0.2.0              27290287067a        6 weeks ago         412MB
```

The final image is including all of the build dependencies. It could probably be reduced in size significantly by using multistage builds and only including runtime deps in the resulting image.

It also isn't checksumming any external files it pulls in, which can make the images non-deterministic and hard to verify.

See https://github.com/getumbrel/umbrel/tree/master/apps#a-good-dockerfile for how to resolve this.

**Comment by louneskmt** [comment]
Thanks, will try to improve this for the next version. 

The dojo-whirlpool image is also 600 MB, but I used their Dockerfile. Maybe I will modify it to make a smaller final image.

**Comment by arcbtc** [comment]
Wow, ace work:

- Ditch "Developer website" as we are not really maintaining it, directing folks to github repo is best
-  Add https://t.me/lnbits to support links
-  Keep amilk disabled, its too buggy/resource intensive to have easy access to
- One line description. Be nice to get the extensions in there "Multi-user and wallet system for Lightning Network, with extension system for adding additional functionality such as LNURL"

**Comment by arcbtc** [comment]
Only if I am not too late that is!

**Comment by mayankchhabra** [comment]
> Only if I am not too late that is!

You're in luck, we just realized that it's Valentines so we're delaying the release by a day lol.

The changes sound good. Regarding the one-line description, there's a 50 character limit on it. I know you can't really fit in the the entire description in 50 characters, but think of it more like a tagline for LNbits, rather than a description. Here's where it shows up in the app store:

Screen Shot 2021-02-14 at 4 52 20 PM

Currently it's set to _"Multi-user wallet management system"_, but let us know if you'd like to change it to something else.

File: apps/lnbits/docker-compose.yml (hunk 1 above)
Comment (arcbtc) [reply apps/lnbits/docker-compose.yml]: Keep it disabled until we manage to improve it for future LNbits release

**Comment by arcbtc** [comment]
Oh yeah, that's nice 👍, leave it like that

**Pull request merged**